test/jdk/javax/sql/resourceBundleTests/ValidateGetBundle.java

```
@@ -0,0 +1,73 @@
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import com.sun.rowset.JdbcRowSetResourceBundle;
+
+import java.util.Locale;
+import java.util.MissingResourceException;
+import java.util.PropertyResourceBundle;
+import java.util.ResourceBundle;
+
+/**
+ * @test
+ * @bug 8294989
+ * @summary Check JDBC RowSet resource bundle access
+ * @modules java.sql.rowset/com.sun.rowset
+ * @run main/othervm --add-opens java.sql.rowset/com.sun.rowset=ALL-UNNAMED ValidateGetBundle
+ */
+public class ValidateGetBundle{
+
+    // Resource bundle base name via a fully qualified class name
+    private static final String FULLY_QUALIFIED_CLASS_NAME = "com.sun.rowset.RowSetResourceBundle";
+    // Resource bundle base name via a path
+    private static final String PATH_TO_BUNDLE = "com/sun/rowset/RowSetResourceBundle";
+    // Default Locale
+    private static final Locale DEFAULT_LOCALE = Locale.getDefault();
+
+    public static void main(String[] args) {
+
+        // The resource bundle should be found with the fully qualified class name
+        testResourceBundleAccess(FULLY_QUALIFIED_CLASS_NAME, true);
+        // The resource bundle will not be found when the path is specified
+        testResourceBundleAccess(PATH_TO_BUNDLE, false);
+    }
+
+    /**
+     * Test to validate whether the JDBC RowSet Resource bundle can be found
+     * @param bundleName the base name of the resource bundle
+     * @param expectBundle indicates whether the resource bundle should be found
+     */
+    private static void testResourceBundleAccess(String bundleName, boolean expectBundle) {
+        try {
+            var bundle = (PropertyResourceBundle) ResourceBundle.getBundle(
+                    bundleName, DEFAULT_LOCALE, JdbcRowSetResourceBundle.class.getModule());
+            System.out.printf("$$$ %s found as expected!%n", bundleName);
```

I think `expectBundle` needs to be examined here. If it is `false`, it should throw an exception.

---

Added a check there for case when expectBundle is false